aiourlshortener/shorteners/base.py

```diff
@@ -35,10 +39,8 @@ def expand(self, url: str) -> str:
 
     @coroutine
     def close(self):
-        try:
+        if self._session is not None:
```

Comment:
tried removing `try` `except` but don't know why it throws `TypeError` so you also need to have this `try` `except` block along with None check

Reply:
What should cause a `TypeError` ?

Reply:
it throws `TypeError: 'NoneType' object is not iterable` with python version 3.4.3
but it works perfectly fine with python version 3.6.1

-----------------
Exact exception on console:
Traceback (most recent call last):
  File "shorturl.py", line 18, in <module>
    loop.run_until_complete(main())
  File "/usr/lib/python3.4/asyncio/base_events.py", line 316, in run_until_complete
    return future.result()
  File "/usr/lib/python3.4/asyncio/futures.py", line 275, in result
    raise self._exception
  File "/usr/lib/python3.4/asyncio/tasks.py", line 236, in _step
    result = coro.send(value)
  File "shorturl.py", line 10, in main
    short_url = yield from shortener.short(url)
  File "/home/ubuntu/aiourlshortener/aiourlshortener/shorteners/__init__.py", line 77, in short
    yield from instance.close()
  File "/home/ubuntu/aiourlshortener/aiourlshortener/shorteners/base.py", line 40, in close
    yield from self._session.close()
TypeError: 'NoneType' object is not iterable

Reply:
it throws `TypeError: 'NoneType' object is not iterable` with python version 3.4.3
but it works perfectly fine with python version 3.6.1

-----------------
Exact exception on console:
Traceback (most recent call last):
  File "shorturl.py", line 18, in <module>
    loop.run_until_complete(main())
  File "/usr/lib/python3.4/asyncio/base_events.py", line 316, in run_until_complete
    return future.result()
  File "/usr/lib/python3.4/asyncio/futures.py", line 275, in result
    raise self._exception
  File "/usr/lib/python3.4/asyncio/tasks.py", line 236, in _step
    result = coro.send(value)
  File "shorturl.py", line 10, in main
    short_url = yield from shortener.short(url)
  File "/home/ubuntu/aiourlshortener/aiourlshortener/shorteners/__init__.py", line 77, in short
    yield from instance.close()
  File "/home/ubuntu/aiourlshortener/aiourlshortener/shorteners/base.py", line 40, in close
    yield from self._session.close()
TypeError: 'NoneType' object is not iterable

Reply:
Which OS with support is shipping `python3.4.3` _(release in 2015/02)_ ?

The old python3.4 version that is installed on my CentOS 7 box with `epel` is `python3.4.5`.

`python3.4.5` does not raise a `TypeError` for calling `yield from session.close()`